projects/01-rubygems-cli/solution/lib/api/validator.rb

```diff
@@ -0,0 +1,15 @@
+class Validator
+  class << self
+    def gem_name_valid?(gem_name)
+      return false if gem_name.nil? or gem_name.empty?
+
+      true
+    end
+
+    def keyword_valid?(keyword)
+      return false if keyword.nil? or keyword.empty?
+
+      true
+    end
```

Comment:
These have the exact same behavior, why not combine them into a single `valid?` or `invalid?` method?

For extra points try rewriting it on a single line, something along the lines of:
```ruby
def self.invalid? = ...
```

Reply:
Also, don't use `or` or `and` in Ruby. They have different operator precedence and can lead to some hard-to-trace bugs.

this method, if `.blank?` is defined could just be `keyword.blank?`
otherwise, this method could be defined as `keyword.nil? || keyword.empty?`